dandi/files/bases.py

```diff
@@ -469,6 +480,31 @@ def get_validation_errors(
         else:
             # make sure that we have some basic metadata fields we require
             try:
+                # Ensure latest version of NWB Inspector is installed and used client-side
+                got_inspector_version = True
+                try:
+                    current_version, max_version = self._get_inspector_versions()
+                except Exception as e:  # In case of no internet connection or other error
+                    got_inspector_version = False
+                    lgr.warning(
+                        "Failed to retrieve NWB Inspector version.",
+                        self.filepath,
+                        type(e).__name__,
+                        str(e),
```

Comment:
These arguments are just going to be ignored, as there are no placeholders in the log message to display them.  Did you mean to include format placeholders?

Reply:
Oh, sorry - copied/pasted from https://github.com/catalystneuro/dandi-cli/blob/force_usage_of_latest_nwb_inspector_release/dandi/files/bases.py#L446-L451 and didn't look very closely - I had assumed they were extra arguments to some kind of logger warning class that captured type/message as separate properties rather than just a single string.

Updated it to be a proper format-string now.

Reply:
You should use `%s` placeholders instead of f-strings for log messages, in order to avoid an error if a stringified exception contains a `%`.

Reply:
Can you give an example of that?

Reply:
> The code you linked to in your previous comment does that; note the %s's in the first argument to log.warning().

I mean, can you provide an example of the justification for

> instead of f-strings for log messages, in order to avoid an error if a stringified exception contains a %.

Reply:
If `str(e)` equals, say, `"Bad format string: '%s %q'"`, then using it in an f-string will cause it to be a substring of the first argument of `log.warning()`, and so `log.warning()` will try to fill in the `%s` with an argument that isn't there and/or choke on the `%q`.  Yes, this is an unlikely scenario in this particular instance, but it costs you nothing to do it the correct, foolproof way from the start.